lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -54,225 +54,218 @@ typedef uint64_t uint64;
 #include <unicode/decimfmt.h>
 #pragma warning(pop)
 
+#include "CommonDefines.h" // INTL_ICU_DEBUG
+
+#if defined(__GNUC__) || defined(__clang__)
+// Output gets included through some other path on Windows
+class Output {
+public:
+    static size_t __cdecl Print(const char16 *form, ...);
+};
+#endif
+
+#define ICU_ERROR_FMT _u("INTL: %S failed with error code %S\n")
+#define ICU_EXPR_FMT _u("INTL: %S failed expression check %S\n")
+
+#ifdef INTL_ICU_DEBUG
+#define ICU_DEBUG_PRINT(fmt, msg) if (INTL_ICU_DEBUG) Output::Print(fmt, __func__, (msg))
```

Comment:
I'd suggest going with 
```cpp
#if defined(INTL_ICU_DEBUG) && INTL_ICU_DEBUG
#define ICU_DEBUG_PRINT(fmt, msg) Output::Print(fmt, __func__, (msg))
#else
#define ICU_DEBUG_PRINT(fmt, msg)
#endif
```

Reply:
Agreed. Keep macro logic in macros.

Also there's not currently and not intended to be any distinction between truthy INTL_ICU_DEBUG versus just being defined in the first place so you can keep the first line as just `#ifdef INTL_ICU_DEBUG` -- unless @jackhorton is intentionally adding a distinction here with the intent of changing the `#define` value of `INTL_ICU_DEBUG` to 0 while keeping it `#define`'d ?

Reply:
Nope, I just brainfarted and left the if() in. Jimmy and I had discussed this offline and I hadn't intended to keep it.